File: AkitaModel/Helpers/AkitaJsonOptions.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace Skyware.Lis.AkitaModel.Helpers;

/// <summary>
/// Standard JSON options for Akita.
/// </summary>
public class AkitaJsonOptions
{

    private static JsonSerializerOptions _Options;

    /// <summary>
    /// JSON options for Akita.
    /// </summary>
    public static JsonSerializerOptions Options => _Options ??= new()
    {
        WriteIndented = true, //TODO: Remove in production
        PropertyNameCaseInsensitive = true,
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
        DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
        ReferenceHandler = ReferenceHandler.IgnoreCycles,
        NumberHandling = JsonNumberHandling.AllowReadingFromString,
    };

}

File: AkitaMotelTS/src/Helpers/AkitaJsonOptions.ts
/**
 * Skyware.Lis.AkitaModel.Helpers
 */

/**
 * Standard JSON options for Akita.
 */
export class AkitaJsonOptions {
  /**
   * JSON serialization options for Akita.
   */
  static getOptions(): any {
    return {
      // TypeScript/JavaScript equivalent settings for JSON serialization
      // These would typically be used with JSON.stringify replacer/reviver functions
      writeIndented: true,
      propertyNameCaseInsensitive: true,
      camelCase: true,
      ignoreNull: true,
      handleCircularReferences: true,
      allowNumbersAsStrings: true
    };
  }
}
